content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+Because authorization flows are security-critical, any caching of identity resolution should choose cache lifetimes carefully. Cache lifetimes of less than 10 minutes are recommended for auth flows specifically.
+
+The resolved DID should be bound to the overall auth session and should be used as the primary account identifier within client app code. Handles (when verified) are acceptable to display in user interfaces, but may change over time and need to be re-verified periodically. When passing an account identifier through to the Authorization Server as part of the Authorization Request in the `login_field`, it is recommended to use the exact account identifier supplied by the user (handle or DID) to ensure any sign-in flow is consistent (users might not recognize their own account DID).
+
+At the end of the auth flow, when the client does an initial token fetch, the Authorization Server must return the account DID in the `sub` field of the JSON response body. If the entire auth flow started with an account identifier, it is critical for the client to verify that this DID matches the expected DID bound to the session earlier; the linkage from account to Authorization Server will already have been verified in this situation.
+
+If the auth flow instead starts with a server (hostname or URL), the client will first attempt to fetch Resource Server metadata (and resolve to Authorization Server if found) and then attempt to fetch Authorization Server metadata. See “Authorization Server” section for server metadata fetching. If either is successful, the client will end up with an identified Authorization Server. The Authorization Request and flow will proceed without a `login_hint` or account identifier being bound to the session, but the Authorization Server `issuer` will be bound to the session.
+
+After the auth flow continues and an initial token request succeeds, the client will parse the account identifier from the `sub` field in the token response. At this point, the client still cannot trust that it has actually authenticated the indicated account. It is critical for the client to resolve the identity (DID document), extract the declared PDS host, and confirm that the PDS (Resource Server) resolves to the Authorization Server bound to the session by fetching the Resource Server metadata.
+
+To reiterate, it is critical for all clients - including those only interested in atproto Identity Authentication - to go through the entire Authorization flow and to verify that the account identifier (DID) in the `sub` field of the token response is consistent with the Authorization Server hostname/origin (`issuer`).
```

Comment:
It is for authN that the verification is really critical. If the `sub` is not used at all (besides for determing the PDS URL) then there is no real risk as it will be the responsibility of the PDS to reject the token.

```suggestion
To reiterate, it is critical for all clients - including those only interested in atproto Authorization - to go through the entire Authorization flow and to verify that the account identifier (DID) in the `sub` field of the token response is consistent with the Authorization Server hostname/origin (`issuer`).
```

or

```suggestion
To reiterate, it is critical for all clients - especially those only interested in atproto Identity Authentication - to go through the entire Authorization flow and to verify that the account identifier (DID) in the `sub` field of the token response is consistent with the Authorization Server hostname/origin (`issuer`).
```

Reply:
Again, a client MUST etc.

And if this is a MUST then you don't need to specify who its especially a MUST for, it's either MUST or it's not. Starting to carve out a special interest even though everyone should do it muddies the waters.

Reply:
I added this section because it would be tempting for folks to just skip it. Everything will "just work" without this check in most cases. These specs are not (yet) IETF RFCs and don't (yet) have the same cultural expectations and editing norms.

I think changing the language to "especially" makes it feel like maybe the other kind of client can skip it, in a way that "including" doesn't.

Reply:
It is just that the current wording seem to indicate that if a client is "only interested in Identity Authentication", it might be less important for them to check the sub. But that is not the case at all!

If we are going to keep some wording that indicates that one use case might be less critical to the verification of the sub, it should probably emphasis on the importance for the right use case.

```suggestion
To reiterate, it is critical for all clients - including those not interested in atproto Identity Authentication - to go through the entire Authorization flow and to verify that the account identifier (DID) in the `sub` field of the token response is consistent with the Authorization Server hostname/origin (`issuer`).
```

Reply:
tracking here: https://github.com/bluesky-social/atproto-website/issues/339